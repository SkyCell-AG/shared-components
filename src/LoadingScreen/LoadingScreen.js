import React from 'react'

import useStyles from './LoadingScreen.styles'

function LoadingScreen() {
    const classes = useStyles()

    return (
        <div className={classes.screen}>
            <div className={classes.center}>
                <div className={classes.logo}>
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        viewBox="0 0 49.17 54.98"
                        width="128"
                        alt="logo"
                    >
                        <title>Logo_Skycell_picto</title>
                        <rect
                            className={classes.white}
                            width="49.17"
                            height="54.98"
                        />
                        <path
                            className={classes.blue}
                            d="M40.43,22.39V17.66L24.3,8.73l-16.1,9,0,19.6L24.31,46.2l16.1-9V22.39ZM32.64,32.53l-8.33,4.63L16,32.54V22.4l8.33-4.63,8.36,4.62Z"
                            transform="translate(0.09 0.25)"
                        />
                        <polygon
                            className={classes.red}
                            points="32.72 32.78 42.91 32.79 42.91 22.63 32.75 22.64 32.72 32.78"
                        />
                        <polygon
                            className={classes.white}
                            points="41.46 28.95 41.45 26.47 39.07 26.47 39.07 24.09 36.59 24.09 36.59 26.47 34.2 26.47 34.2 28.95 36.59 28.95 36.59 31.34 39.07 31.34 39.07 28.95 41.46 28.95"
                        />
                    </svg>
                </div>
                <div className={classes.spinnerWrapper}>
                    <div className={classes.spinner}>
                        <div className={classes.inner}>
                            <div className={classes.gap} />
                            <div className={classes.left}>
                                <div className={classes.halfCircleLeft} />
                            </div>
                            <div className={classes.right}>
                                <div className={classes.halfCircleRight} />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default React.memo(LoadingScreen)
