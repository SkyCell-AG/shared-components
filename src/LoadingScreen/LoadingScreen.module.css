.screen {
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ffffff;
    z-index: 99999;
    pointer-events: none;
}

.center {
    display: block;
    width: 100%;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
}

.logo {
    width: 128px;
    margin: 0 auto;
}

.spinnerWrapper {
    display: block;
    position: relative;
    width: 100%;
    min-height: 100px;
    height: 100px;
}

.spinner {
    position: absolute;
    overflow: hidden;
    left: 50%;
    margin-left: -50px;
    animation: outerRotate 2.91667s linear infinite;
}

.inner {
    width: 100px;
    height: 100px;
    position: relative;
    animation: sporadicRotate 5.25s cubic-bezier(0.35, 0, 0.25, 1) infinite;
}

.gap {
    position: absolute;
    left: 49px;
    right: 49px;
    top: 0;
    bottom: 0;
    border-top: 10px solid;
    box-sizing: border-box;
}

.left {
    position: absolute;
    top: 0;
    height: 100px;
    width: 50px;
    overflow: hidden;
    left: 0;
}
.right {
    position: absolute;
    top: 0;
    height: 100px;
    width: 50px;
    overflow: hidden;
    right: 0;
}

.halfCircleLeft {
    left: 0;
    animation: leftWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite;
    -webkit-animation: leftWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite;
    position: absolute;
    top: 0;
    width: 100px;
    height: 100px;
    box-sizing: border-box;
    border: 10px solid #6FCAE9;
    border-bottom-color: transparent;
    border-right-color: transparent;
    border-radius: 50%;
}

.halfCircleRight {
    right: 0;
    animation: rightWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite;
    -webkit-animation: rightWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite;
    position: absolute;
    top: 0;
    width: 100px;
    height: 100px;
    box-sizing: border-box;
    border: 10px solid #6FCAE9;
    border-bottom-color: transparent;
    border-left-color: transparent;
    border-radius: 50%;
}

@keyframes outerRotate {
    0% {
        transform: rotate(0deg) scale(0.5);
    }

    100% {
        transform: rotate(360deg) scale(0.5);
    }
}

@keyframes leftWobble {

    0%,
    100% {
        transform: rotate(130deg);
    }

    50% {
        transform: rotate(-5deg);
    }
}

@keyframes rightWobble {

    0%,
    100% {
        transform: rotate(-130deg);
    }

    50% {
        transform: rotate(5deg);
    }
}

@keyframes sporadicRotate {
    12.5% {
        transform: rotate(135deg);
    }

    25% {
        transform: rotate(270deg);
    }

    37.5% {
        transform: rotate(405deg);
    }

    50% {
        transform: rotate(540deg);
    }

    62.5% {
        transform: rotate(675deg);
    }

    75% {
        transform: rotate(810deg);
    }

    87.5% {
        transform: rotate(945deg);
    }

    100% {
        transform: rotate(1080deg);
    }
}

.white {
    fill:#fff;
}
.blue {
    fill:#62c6e9;
}
.red {
    fill:#e63329;
}