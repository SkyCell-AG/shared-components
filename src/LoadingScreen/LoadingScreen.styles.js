import {
    makeStyles,
} from '@mui/styles'

const useStyles = makeStyles(() => {
    return {
        '@keyframes outerRotate': {
            '0%': {
                transform: 'rotate(0deg) scale(0.5)',
            },

            '100%': {
                transform: 'rotate(360deg) scale(0.5)',
            },
        },

        '@keyframes leftWobble': {

            '0%': {},
            '100%': {
                transform: 'rotate(130deg)',
            },

            '50%': {
                transform: 'rotate(-5deg)',
            },
        },

        '@keyframes rightWobble': {

            '0%': {},
            '100%': {
                transform: 'rotate(-130deg)',
            },

            '50%': {
                transform: 'rotate(5deg)',
            },
        },

        '@keyframes sporadicRotate': {
            '12.5%': {
                transform: 'rotate(135deg)',
            },

            '25%': {
                transform: 'rotate(270deg)',
            },

            '37.5%': {
                transform: 'rotate(405deg)',
            },

            '50%': {
                transform: 'rotate(540deg)',
            },

            '62.5%': {
                transform: 'rotate(675deg)',
            },

            '75%': {
                transform: 'rotate(810deg)',
            },

            '87.5%': {
                transform: 'rotate(945deg)',
            },

            '100%': {
                transform: 'rotate(1080deg)',
            },
        },
        screen: {
            display: 'block',
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            bottom: 0,
            backgroundColor: '#ffffff',
            zIndex: 99999,
            pointerEvents: 'none',
        },

        center: {
            display: 'block',
            width: '100%',
            position: 'absolute',
            top: '50%',
            transform: 'translateY(-50%)',
        },

        logo: {
            width: 128,
            margin: '0 auto',
        },

        spinnerWrapper: {
            display: 'block',
            position: 'relative',
            width: '100%',
            minHeight: 100,
            height: 100,
        },

        spinner: {
            position: 'absolute',
            overflow: 'hidden',
            left: '50%',
            marginLeft: -50,
            animation: '$outerRotate 2.91667s linear infinite',
        },

        inner: {
            width: 100,
            height: 100,
            position: 'relative',
            animation: '$sporadicRotate 5.25s cubic-bezier(0.35, 0, 0.25, 1) infinite',
        },

        gap: {
            position: 'absolute',
            left: 49,
            right: 49,
            top: 0,
            bottom: 0,
            borderTop: '10px solid',
            boxSizing: 'border-box',
        },

        left: {
            position: 'absolute',
            top: 0,
            height: 100,
            width: 50,
            overflow: 'hidden',
            left: 0,
        },
        right: {
            position: 'absolute',
            top: 0,
            height: 100,
            width: 50,
            overflow: 'hidden',
            right: 0,
        },

        halfCircleLeft: {
            left: 0,
            animation: '$leftWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite',
            '-webkit-animation': '$leftWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite',
            position: 'absolute',
            top: 0,
            width: 100,
            height: 100,
            boxSizing: 'border-box',
            border: '10px solid #6FCAE9',
            borderBottomColor: 'transparent',
            borderRightColor: 'transparent',
            borderRadius: '50%',
        },

        halfCircleRight: {
            right: 0,
            animation: '$rightWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite',
            '-webkit-animation': '$rightWobble 1.3125s cubic-bezier(0.35, 0, 0.25, 1) infinite',
            position: 'absolute',
            top: 0,
            width: 100,
            height: 100,
            boxSizing: 'border-box',
            border: '10px solid #6FCAE9',
            borderBottomColor: 'transparent',
            borderLeftColor: 'transparent',
            borderRadius: '50%',
        },
        white: {
            fill: '#fff',
        },
        blue: {
            fill: '#62c6e9',
        },
        red: {
            fill: '#e63329',
        },
    }
})

export default useStyles
