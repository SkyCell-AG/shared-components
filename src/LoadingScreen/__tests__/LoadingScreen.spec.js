import React from 'react'
import {
    shallow,
} from 'enzyme'

import LoadingScreen from '../LoadingScreen'

describe('app/Apps/shared-components/LoadingScreen', () => {
    it('LoadingScreen page snapshot', () => {
        const wrapper = shallow(
            <LoadingScreen />,
        )

        expect(wrapper).toMatchSnapshot()
    })
})
