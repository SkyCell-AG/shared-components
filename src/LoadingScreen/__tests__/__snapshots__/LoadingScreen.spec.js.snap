// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/Apps/shared-components/LoadingScreen LoadingScreen page snapshot 1`] = `
<div
  className="makeStyles-screen-5"
>
  <div
    className="makeStyles-center-6"
  >
    <div
      className="makeStyles-logo-7"
    >
      <svg
        alt="logo"
        viewBox="0 0 49.17 54.98"
        width="128"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Logo_Skycell_picto
        </title>
        <rect
          className="makeStyles-white-16"
          height="54.98"
          width="49.17"
        />
        <path
          className="makeStyles-blue-17"
          d="M40.43,22.39V17.66L24.3,8.73l-16.1,9,0,19.6L24.31,46.2l16.1-9V22.39ZM32.64,32.53l-8.33,4.63L16,32.54V22.4l8.33-4.63,8.36,4.62Z"
          transform="translate(0.09 0.25)"
        />
        <polygon
          className="makeStyles-red-18"
          points="32.72 32.78 42.91 32.79 42.91 22.63 32.75 22.64 32.72 32.78"
        />
        <polygon
          className="makeStyles-white-16"
          points="41.46 28.95 41.45 26.47 39.07 26.47 39.07 24.09 36.59 24.09 36.59 26.47 34.2 26.47 34.2 28.95 36.59 28.95 36.59 31.34 39.07 31.34 39.07 28.95 41.46 28.95"
        />
      </svg>
    </div>
    <div
      className="makeStyles-spinnerWrapper-8"
    >
      <div
        className="makeStyles-spinner-9"
      >
        <div
          className="makeStyles-inner-10"
        >
          <div
            className="makeStyles-gap-11"
          />
          <div
            className="makeStyles-left-12"
          >
            <div
              className="makeStyles-halfCircleLeft-14"
            />
          </div>
          <div
            className="makeStyles-right-13"
          >
            <div
              className="makeStyles-halfCircleRight-15"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
